src/test/fuzz/coins_view.cpp

```diff
@@ -0,0 +1,295 @@
+            break;
+        }
+        case 4: {
+            coins_view_cache.Uncache(random_out_point);
+            break;
+        }
+        case 5: {
+            if (fuzzed_data_provider.ConsumeBool()) {
+                backend_coins_view = CCoinsView{};
+            }
+            coins_view_cache.SetBackend(backend_coins_view);
+            break;
+        }
+        case 6: {
+            const std::optional<COutPoint> opt_out_point = ConsumeDeserializable<COutPoint>(fuzzed_data_provider);
+            if (!opt_out_point) {
+                break;
+            }
+            random_out_point = *opt_out_point;
+            break;
+        }
+        case 7: {
+            const std::optional<Coin> opt_coin = ConsumeDeserializable<Coin>(fuzzed_data_provider);
+            if (!opt_coin) {
+                break;
+            }
+            random_coin = *opt_coin;
+            break;
+        }
+        case 8: {
+            const std::optional<CMutableTransaction> opt_mutable_transaction = ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider);
+            if (!opt_mutable_transaction) {
+                break;
+            }
+            random_mutable_transaction = *opt_mutable_transaction;
+            break;
+        }
+        case 9: {
+            CCoinsMap coins_map;
+            while (fuzzed_data_provider.ConsumeBool()) {
+                CCoinsCacheEntry coins_cache_entry;
+                coins_cache_entry.flags = fuzzed_data_provider.ConsumeIntegral<unsigned char>();
+                if (fuzzed_data_provider.ConsumeBool()) {
+                    coins_cache_entry.coin = random_coin;
+                } else {
+                    const std::optional<Coin> opt_coin = ConsumeDeserializable<Coin>(fuzzed_data_provider);
+                    if (!opt_coin) {
+                        break;
+                    }
+                    coins_cache_entry.coin = *opt_coin;
+                }
+                if (fuzzed_data_provider.ConsumeBool()) {
+                    const std::optional<COutPoint> opt_out_point = ConsumeDeserializable<COutPoint>(fuzzed_data_provider);
+                    if (!opt_out_point) {
+                        break;
+                    }
+                    random_out_point = *opt_out_point;
+                }
+                coins_map.emplace(random_out_point, std::move(coins_cache_entry));
+            }
+            bool expected_code_path = false;
+            try {
+                coins_view_cache.BatchWrite(coins_map, fuzzed_data_provider.ConsumeBool() ? ConsumeUInt256(fuzzed_data_provider) : coins_view_cache.GetBestBlock());
+                expected_code_path = true;
+            } catch (const std::logic_error& e) {
+                if (e.what() == std::string{"FRESH flag misapplied to coin that exists in parent cache"}) {
+                    expected_code_path = true;
```

Comment:
Note to myself only: I thought about putting a more strict assert here, but the exception doesn't give any hints which coin failed, so I don't think this is possible for now.